src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -2887,13 +2887,7 @@ public async Task UploadStreamItemInvalidType()
         [Fact]
         public async Task UploadStreamItemInvalidId()
         {
-            bool ExpectedErrors(WriteContext writeContext)
-            {
-                return writeContext.LoggerName == "Microsoft.AspNetCore.SignalR.HubConnectionHandler" &&
-                       writeContext.EventId.Name == "ErrorProcessingRequest";
-            }
-
-            using (StartVerifiableLog(ExpectedErrors))
+            using (StartVerifiableLog())
```

Comment:
Does this test verify "StreamItemMessage received unexpectedly." got logged?

Reply:
No, the verifiable log stuff checks that no unexpected `Error` logs appear in the test.

These tests will be better once we add the stream completion, because we'll verify that the completion was sent from the server on failure.

Reply:
It's good that you're verifying there are no unexpected errors, but you should also verify "StreamItemMessage received unexpectedly." was logged so you know you are hitting the code path you are trying to test.

Reply:
We know we're hitting the code path because the connection isn't being closed anymore.

We generally don't test log messages, I did it before because I couldn't think of anything better.

Reply:
I understand not wanting to test logs. If you're not careful, it can definitely make tests unnecessarily fragile.

That being said, this PR description claims "We should instead ignore (and log) any stream activity after it has been closed", and I think if that's a major goal of this PR, we should verify it.

The tests these are replacing included lines like `Assert.True(errorLogged);` anyway, so it's not like we weren't testing the logs before. The old tests were also verifying the exception message matched exactly what was expected. Sharing resource strings with the test project could help make the test less fragile.